### DISCUSSION TITLE
Dispatch is not working as expected.

### DISCUSSION THREAD

**Discussion by StaffCollab** [filamentphp/filament#14759] [Help]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

Here is my component. (Please understand that I need to be able to submit via an enter-keydown this specific form field and not the complete form and that's why I had to add the extra alpine attribute
~~~
 TextInput::make('message')
               ->extraAlpineAttributes([
                    'x-ref' => 'message',
                    'x-on:keydown.enter.prevent' => '$dispatch("addMessage", $refs.message.value)',
                ])

~~~

And in the resource, I created this method

~~~
    #[On('addMessage')]
    public function addMessage($message)
    {
        dump('addMessage');
    }
~~~

But it is not being called.

In the browser console, I checked like this:

~~~

window.addEventListener('addMessage', function(event) {
    console.log('addMessage event dispatched:', event.detail);
});

~~~
And the event is getting dispatched as a browser event, but I cannot figure out why I can't listen to it with the #[On()] Attribute

**Comment by leandrocfe** [comment]
You should add `addMessage` method to the Pages, CreatePage.php, EditPage.php. If you are using a resource simple, you need to add this to the ManagePage.php.
